### ISSUE TITLE
Replace Content inside scanned Page

### ISSUE THREAD

**Issue by test1964** [andybalholm/redwood#16]
Hey,

Is there a way to replace html code inside scanned page?
For now we have the "pruning.conf" that can only remove from page.
For example can it do like  Dansguardian (contentregexplist):
<html> -> <script language='javascript'>....some code....<html>
Or:
<td> .*car</td>  -> <td> new Text</td>

**Comment by test1964** [comment]
Sorry but the Example (with html code) not visible very good will write again with spaces:
1. H T M L  ->  <script language='javascript'>....some code.... </script> H T M L
2. < T D > new .*car < / T D > -> < T D > new text < / T D >

**Comment by thinkwelltwd** [comment]
No, there's not a way to do this; there are more ways to get this wrong than right. Dansguardian  got it wrong, in that there was no way to limit the substitution to specific hosts. This tended to indiscriminately break pages. Also, it was one of the most resource-intensive features of Dansguardian. 

Having said all that, could you elaborate more on the use-cases / pain points behind your inquiry?

**Comment by test1964** [comment]
Dansguardian was just example about the issue, Examples for using it: 
- cnn.com - want to remove some headlines(with small pics) based on text search(let say there is a link to some rape story)
- some sites when you remove some content(with some html code), then you  need to put another code so page will still be visible well

**Comment by thinkwelltwd** [comment]
The cnn.com example is already possible, as it involves only removal and not substitution. Pruning  removes elements from a page by CSS selector instead of regex. 

Optional pruning is possible in two ways - 

1. Adding a score to the prune rule does phrase filtering `cnn.com 30 div#headlines`
2. Jquery-like "contains" `cnn.com a:contains('rape')

Could you provide a specific site / page where substitution of DOM elements is required to preserve page functionality, or better yet, a page where the current pruning functionality is not adequate?

**Comment by test1964** [comment]
For example in the site  "http://news.sky.com", I want to remove some headline in the right with the picture,
when I try all page get mass.
Can You give an example how to handle this - Most div inside without id and some class with more then one word.
I try to remove this:
div  class="sdc-news-story-grid__card sdc-news-story-grid__card--3col   " data-id="10809235"
based on data-id without success(how to it?)

**Comment by thinkwelltwd** [comment]
>I try to remove this:
div class="sdc-news-story-grid__card sdc-news-story-grid__card--3col " data-id="10809235"
based on data-id without success(how to it?)

`sky.com div[data-id="10809235"]`

To remove the entire headlines side bar:

`sky.com article#component-highlights`

To remove sidebar articles based a phrase filtered score of content inside the object exceeds 50:

`sky.com 50 div.sdc-news-story-grid__card`

To remove sidebar articles that contain a specific phrase, such as "Trump":
`sky.com div.sdc-news-story-grid__card:contains("Trump")`

In summary, the pruning rules Redwood offers are dramatically more powerful than regex-based substitutions from Dansguardian. I'd recommend to forget about regexes and get CSS selectors down cold, and you'll do far more than Dansguardian enabled you to accomplish.

**Comment by test1964** [comment]
Thanks for all examples.
 I checked and bow understand how to use it - works very well.
Have only one problem how to handle this issue:
Block all page if in title I have the word "testing"(<title> this is testing </title>),I try 
     example.com   html:contains('testing')
but it not working,how to fix it?

**Comment by andybalholm** [comment]
So you want the page to become blank, rather than blocking properly?

**Comment by test1964** [comment]
In this case , yes.,
can it be done?

**Comment by andybalholm** [comment]
I don't know why it wouldn't work. Can you give me the real example that isn't working, instead of example.com?

**Comment by test1964** [comment]
Here real example:
In pruning.conf have this line
cnn.com   html:contains('Video News</title>')

When I try the main cnn.com site I still get the page 

can you check why?

**Comment by andybalholm** [comment]
The :contains selector matches only the text content of the element, not its innerHTML.

**Comment by test1964** [comment]
Then how to replace all html code based on words inside title?

**Comment by thinkwelltwd** [comment]
>Then how to replace all html code based on words inside title?

As I've tried to explain before, Redwood doesn't do page substitution, and I've supplied examples why it doesn't need to. This latest scenario is not a valid case for contentregexplist in Dansguardian, nor pruning in Redwood.  

Horses for Courses; Tools for the Job. 

Pruning for page modification. Page blocking for blocking the whole page. 

If you want to block the page based on the page title, it's as simple as entering this rule in a Blocked category:

&lt;page title text&gt; 105

And the user will be shown the blocked page, assuming the threshold of 100, that you've suggested earlier.

**Comment by test1964** [comment]
Thanks
